astropy/units/distribution.py

```diff
@@ -0,0 +1,427 @@
+    def pdf_smad(self):
+        """
+        The median absolute deviation of this distribution rescaled to match the
+        standard deviation for a normal distribution.
+        """
+        return self.pdf_mad * self._smad_scale_factor
+
+    def percentiles(self, perc, **kwargs):
+        """
+        Compute percentiles of this Distribution.
+
+        Parameters
+        ----------
+        perc : float or array of floats
+            The desired  precentiles of the distribution (i.e., on [0,100]).
+        kwargs
+            Additional keywords are passed into `numpy.percentile`.
+
+        Returns
+        -------
+        percs : Quantity
+            The ``fracs`` percentiles of this distribution.
+        """
+        perc = np.percentile(self.distribution, perc, axis=-1)
+        # numpy.percentile strips units for unclear reasons, so we have to make
+        # a new object with units
+        if hasattr(self.distribution, '_new_view'):
+            return self.distribution._new_view(perc)
+        else:
+            return perc
+
+    def hist(self, maxtoshow=10, **kwargs):
+        """
+        Use `astropy.visualization.hist` (which uses matplotlib's ``hist``
+        function) to visualize this distribution.  For N-D distributions, the
+        array is flattened following standard numpy rules, and the distributions
+        are shown as separate histograms for each element.
+
+        Parameters
+        ----------
+        maxtoshow : int or `None`
+            The maximum number of distribution elements to show.  If `None`,
+            there will be no limit, but this may overwhelm matplotlib if the
+            distribution is large.
+        Additional keywords are passed into `astropy.visualization.hist`
+        """
+        # NOTE: not adjusted to new structure!!
+        labelset = 'label' in kwargs
+        scalar_distr = len(self.shape) == 1
+        if scalar_distr:
+            reshaped = [self]
+        else:
+            reshaped = self.reshape(self.n_samples, self.size//self.n_samples).T
+
+        hists = []
+        for i, dat in enumerate(reshaped):
+            if i >= maxtoshow:
+                break
+            if (not scalar_distr) and (not labelset):
+                if len(self.shape) == 2:
+                    idxstr = str(i)
+                else:
+                    idxstr = str(np.unravel_index(i, self.shape[1:]))
+                kwargs['label'] = 'Distr ' + idxstr
+            hists.append(visualization.hist(dat, **kwargs))
+        return hists
+
+
+class NormalDistribution(Distribution):
+    """
+    A Gaussian/normal Distribution
+
+    Parameters
+    ----------
+    center : `Quantity`
+        The center of this `NormalDistribution`
+    std : `Quantity` or `None`
+        The standard deviation/σ of this distribution. Shape must match and unit
+        must be compatible with ``center``, or be `None` (if ``var`` or ``ivar``
+        are set).
+    var : `Quantity` or `None`
```

Comment:
I feel there is too much choice here, just stick with `std` which is most common and what goes into `randr` anyway.

Separately, I wondered about adding a `seed` argument, or alternatively a `RandomState` instance. -- although I guess that can be done with the context anyway.